server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java

```diff
@@ -178,10 +178,10 @@ public String asKey() {
 
     @Override
     protected String mapAddressInternal(String user, Domain domain) throws RecipientRewriteTableException {
-        Mappings mappings = OptionalUtils.or(
+        Mappings mappings = OptionalUtils.executeIfEmpty(
```

Comment:
:-1: I don't find it readable at all.

Reply:
Not only it is false, but also it is buggy ;-) 

executeIfEmpty is a side effect ;-)